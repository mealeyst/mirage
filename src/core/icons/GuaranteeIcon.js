import React from 'react'
import PropTypes from 'prop-types'
import styled, { css, keyframes } from 'styled-components'

const UnstyledGuaranteeIcon = ({className}) => {
  return (
    <svg className={className} xmlns='http://www.w3.org/2000/svg' viewBox='0 0 42 42'>
      <path d='M21,2.555c1.389,0,4.005-1.869,5.347-1.509s2.673,3.286,3.876,3.981,4.4.384,5.385,1.366.672,4.182,1.366,5.385,3.621,2.534,3.981,3.876S39.445,19.611,39.445,21s1.809,3.99,1.509,5.347-3.286,2.673-3.981,3.876-.384,4.4-1.366,5.385-4.182.672-5.385,1.366-2.534,3.621-3.876,3.981S22.389,39.445,21,39.445s-4.005,1.869-5.347,1.509-2.673-3.286-3.876-3.981-4.4-.384-5.385-1.366-.672-4.182-1.366-5.385S1.345,27.7,1.046,26.347,2.555,22.389,2.555,21,.686,16.995,1.046,15.653s3.286-2.673,3.981-3.876.384-4.4,1.366-5.385,4.182-.672,5.385-1.366,2.534-3.621,3.876-3.981S19.611,2.555,21,2.555Z'/>
      <circle cx='21' cy='21' r='14.7'/>
      <polyline points='14.629 21.734 18.691 25.483 26.345 17.203'/>
    </svg>
  )
}

const rotate = keyframes`
  0% {
    transform: rotate(0deg);
  }
  25% {
    transform: rotate(90deg);
  }
  50% {
    transform: rotate(180deg);
  }
  75% {
    transform: rotate(270deg);
  }
  100% {
    transform: rotate(360deg);
  }
`

const animation = css`
  animation: ${rotate};
  animation-duration: 25s;
  animation-iteration-count: infinite;
  transform-origin: 50% 50%;
  animation-timing-function: linear;
`

const GuaranteeIcon = styled(UnstyledGuaranteeIcon)`
  width: ${props => props.width};
  circle {
    fill: ${props => props.theme.colors.white};
  }
  path {
    fill: ${props => props.theme.colors.rocketBlue};
    ${props => props.animated && animation}
  }
  polyline {
    stroke-linecap: round;
    stroke-width: 2;
    fill: none;
    stroke: ${props => props.theme.colors.rocketBlue};
  }
`

GuaranteeIcon.propTypes = {
  width: PropTypes.string.isRequired
}

GuaranteeIcon.defaultProps = {
  width: '2.5rem'
}

/** @component */
export default GuaranteeIcon
