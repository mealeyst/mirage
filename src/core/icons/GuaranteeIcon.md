```js
<GuaranteeIcon />
```

### Animated Example:

```js
<GuaranteeIcon animated />
```
