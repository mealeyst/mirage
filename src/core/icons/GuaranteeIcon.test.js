import React from 'react'
import { css, keyframes } from 'styled-components'
import 'jest-styled-components'

import { GuaranteeIcon } from 'SRC'
import { animation, UnstyledGuaranteeIcon } from './GuaranteeIcon'

const { shallowWithTheme } = global

describe('(Styled Component) GuaranteeIcon', () => {
  const createGuaranteeIcon = (props) => {
    return shallowWithTheme(<GuaranteeIcon {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createGuaranteeIcon())
    .toMatchSnapshot()
  })

  test('setting the width', () => {
    const width = '6rem'
    expect(createGuaranteeIcon({width: width}))
    .toHaveStyleRule({
      width: width
    })
  })

  test('setting the width', () => {
    expect(createGuaranteeIcon({animated: true}))
    .toHaveStyleRule({
      modifer: css`${animation}`
    })
  })
})

describe('(Component) UnstyledGuaranteeIcon', () => {
  const createUnstyledGuaranteeIcon = (props) => {
    return shallowWithTheme(<UnstyledGuaranteeIcon {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createUnstyledGuaranteeIcon())
    .toMatchSnapshot()
  })

  test('setting the className', () => {
    const className = 'example-class'
    expect(
      createUnstyledGuaranteeIcon({className: className}).prop('className')
    ).toEqual(className)
  })
})
