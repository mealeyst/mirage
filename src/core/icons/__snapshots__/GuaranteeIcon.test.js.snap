// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) UnstyledGuaranteeIcon matching the snapshot 1`] = `
<svg
  viewBox="0 0 42 42"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M21,2.555c1.389,0,4.005-1.869,5.347-1.509s2.673,3.286,3.876,3.981,4.4.384,5.385,1.366.672,4.182,1.366,5.385,3.621,2.534,3.981,3.876S39.445,19.611,39.445,21s1.809,3.99,1.509,5.347-3.286,2.673-3.981,3.876-.384,4.4-1.366,5.385-4.182.672-5.385,1.366-2.534,3.621-3.876,3.981S22.389,39.445,21,39.445s-4.005,1.869-5.347,1.509-2.673-3.286-3.876-3.981-4.4-.384-5.385-1.366-.672-4.182-1.366-5.385S1.345,27.7,1.046,26.347,2.555,22.389,2.555,21,.686,16.995,1.046,15.653s3.286-2.673,3.981-3.876.384-4.4,1.366-5.385,4.182-.672,5.385-1.366,2.534-3.621,3.876-3.981S19.611,2.555,21,2.555Z"
  />
  <circle
    cx="21"
    cy="21"
    r="14.7"
  />
  <polyline
    points="14.629 21.734 18.691 25.483 26.345 17.203"
  />
</svg>
`;

exports[`(Styled Component) GuaranteeIcon matching the snapshot 1`] = `
.c0 {
  width: 2.5rem;
}

.c0 circle {
  fill: #FFFFFF;
}

.c0 path {
  fill: #0073D1;
}

.c0 polyline {
  stroke-linecap: round;
  stroke-width: 2;
  fill: none;
  stroke: #0073D1;
}

<UnstyledGuaranteeIcon
  className="c0"
  width="2.5rem"
/>
`;
